sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -406,6 +406,151 @@ def _upload_blob_options(  # pylint:disable=too-many-statements
+                               encryption_algorithm=cpk.algorithm)
+        kwargs['cpk_info'] = cpk_info
+        kwargs['cpk_scope_info'] = get_cpk_scope_info(kwargs)
+
+        options = {
+            'content_length': 0,
+            'copy_source_blob_properties': include_source_blob_properties,
+            'source_content_md5': bytearray(source_content_md5) if source_content_md5 else None,
+            'copy_source': source_url,
+            'timeout': timeout,
+            'modified_access_conditions': dest_mod_conditions,
+            'blob_tags_string': blob_tags_string,
+            'headers': headers,
+            'cls': return_response_headers,
+        }
+        source_mod_conditions = get_source_conditions(kwargs)
+        dest_access_conditions = get_access_conditions(kwargs.pop('destination_lease', None))
+        options['source_modified_access_conditions'] = source_mod_conditions
+        options['lease_access_conditions'] = dest_access_conditions
+        options['tier'] = tier.value if tier else None
+        options.update(kwargs)
+        return options
+
+    @distributed_trace
+    def upload_blob_from_url(self, source_url, **kwargs):
+        # type: (str, Any) -> Dict[str, Any]
+        """
+        Creates a new Block Blob where the content of the blob is read from a given URL.
+        The content of an existing blob is overwritten with the new blob.
+
+        :param str source_url:
+            A URL of up to 2 KB in length that specifies a file or blob.
+            The value should be URL-encoded as it would appear in a request URI.
+            If the source is in another account, the source must either be public
+            or must be authenticated via a shared access signature. If the source
+            is public, no authentication is required.
+            Examples:
+            https://myaccount.blob.core.windows.net/mycontainer/myblob
+
+            https://myaccount.blob.core.windows.net/mycontainer/myblob?snapshot=<DateTime>
+
+            https://otheraccount.blob.core.windows.net/mycontainer/myblob?sastoken
+        :keyword bool include_source_blob_properties:
+            Indicates if properties from the source blob should be copied. Defaults to True.
+        :keyword tags:
+            Name-value pairs associated with the blob as tag. Tags are case-sensitive.
+            The tag set may contain at most 10 tags.  Tag keys must be between 1 and 128 characters,
+            and tag values must be between 0 and 256 characters.
+            Valid tag key and value characters include: lowercase and uppercase letters, digits (0-9),
+            space (` `), plus (+), minus (-), period (.), solidus (/), colon (:), equals (=), underscore (_)
+        :paramtype tags: dict(str, str)
+        :keyword bytearray source_content_md5:
+            Specify the md5 calculated for the range of
+            bytes that must be read from the upload source.
```

Comment:
the "range of bytes" sound a bit confusing. this api doesn't have partial copy capabilities. it should say that this hash is used to verify integrity of copied bytes.